aiven/client/cliarg.py

```diff
@@ -43,6 +43,33 @@ def wrapped(self):
     return wrapper
 
 
+def json_path_or_string_parameter(param_name):
+    """JSON parameter handler"""
+
+    def wrapper(fun):
+        arg(
+            param_name,
+            help="JSON string or path (preceded by '@') to a JSON configuration file",
+        )(fun)
+
+        @wraps(fun)
+        def wrapped(self):
+            sanitized_param_name = param_name.replace("--", "").replace("-", "_")
```

Comment:
Why is this `replace("-", "_")` needed? For example if the param name in the request is `schema_sql`, is the idea here that `schema-sql` gets converted into `schema_sql`? Why wouldn't the caller have `schema_sql` in the first place. That's the name of the parameter that can be seen in the API docs.

What I think that could be the idea is that before this PR the switch to use is `--schema-sql` and thus it's kind of logical to support `schema-sql` in the JSON.

But I don't this should be done, since the name of the parameter in the API is not `schema-sql`.

And why `replace("--", "")`?

Reply:
The `-` to `_` is replaced as it will be the name of the member in the `self.args`.

The replace for `--` is the same as in the `user_config_json` but with different flavor as the parameter name is known there. In here the parameter name is not known and sanitazion is done programmatically.
See here: https://github.com/aiven/aiven-client/blob/a99d7eabe0183e2544aa85d3aa104de3dc1c1856/aiven/client/cliarg.py#L89-L92

What could be done is to move the logic when calling `arg` on line 50. Or change have specialized function for `--table-properties` like what is done with `--user-config-json`.